src/math_utils.rs

```diff
@@ -638,6 +901,35 @@ mod tests {
             }
         }
 
+            #[test]
+             // Test for sqrt_prime_power_ of a quadratic residue using CAIRO_PRIME. Result should be the minimum root.
+             fn sqr_prime_power_using_cairo_prime(ref x in "([1-9][0-9]*)") {
+                 let x = &BigUint::parse_bytes(x.as_bytes(), 10).unwrap();
+                 let x_sq = x * x;
+                 let sqrt = sqrt_prime_power(&x_sq, &CAIRO_PRIME).unwrap_or_default();
+
+                if &sqrt != x {
+                    assert_eq!(Felt252::max_value().to_biguint() - sqrt + 1_usize, *x);
+                } else {
+                    assert_eq!(&sqrt, x);
+                }
+            }
+
+            #[test]
+             // Test for sqrt_prime_power_ of a quadratic residue. Result should be the minimum root.
+             fn sqrt_prime_power_using_random_prime(ref x in "([1-9][0-9]*)", ref y in "([1-9][0-9]*)") {
+                 let x = &BigUint::parse_bytes(x.as_bytes(), 10).unwrap();
+                 let p = &BigUint::parse_bytes(y.as_bytes(), 10).unwrap();
+                 let x_sq = x * x;
+                 let sqrt = sqrt_prime_power(&x_sq, &CAIRO_PRIME).unwrap_or_default();
+
+                if &sqrt != x {
+                    assert_eq!(p - sqrt, *x);
+                } else {
+                    assert_eq!(&sqrt, x);
+                }
+            }
+
```

Comment:
nit: there seems to be an extra level of indentation compared to the other tests.

Reply:
Yes, the proptest macro messes up cargo-fmt, Ill fix it

Reply:
That explains why it happened to Timothee back then.